[CXl0l0] iP

CXl0l0 opened the pull request:
Sigma

> "A true sigma male roams through life as a lone wolf, thriving in solitude." – :wolf:

![An alpha sigma lone wolf.](https://preview.redd.it/skott-canonically-refers-to-himself-as-the-lone-wolf-tf-is-v0-lam0dbwzd2gd1.jpeg?width=572&format=pjpg&auto=webp&s=f53d92a215f06cf704ac382a2f94adbbbce9b4a0)

Wanna be a true sigma and unlock your full potential? Start by using Sigma and start the grind 💪🐺💪 . It's,

- text-based
- easy-peasy lemon squeezy to use
- ~~SUPER~~ **ULTRA SIGMA** FAST to use

Just,

1. download it from [here](https://github.com/CXl0l0/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you
5. **ASCEND TO SIGMA STATUS** 💪 🐺 🚬 

And it is **FREE!** 
😱 😱 🙀 🤯 🤯  🙀 🫨 

Features:



- [ ] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

CXl0l0 committed:
add greetings and exit

CXl0l0 committed:
added echoing

CXl0l0 committed:
added list

CXl0l0 committed:
Added marking done/not done feature for tasks

CXl0l0 committed:
added todo, deadline and event

CXl0l0 committed:
added test using I/O redirection technique

CXl0l0 committed:
added error exceptions

CXl0l0 committed:
improved exception messages to include solutions

CXl0l0 committed:
added function to delete tasks from list

caroline1233456 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+        try {
+            String taskName = "";
+            String date = "";
+            boolean reading_date = false;
```
Should the boolean variable be named to sound like a boolean? I noticed the same issue in several other places too.

mjyan02 replied:
This boolean variable does not sound like a boolean. Perhaps you should consider renaming it? Boolean variables in other files could also use the same edit.

caroline1233456 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+        }
+    }
+
+    private void add_event(String[] tokens) throws SigmaException {
```
Should the name of the method be in camelCase? I noticed the same issue in several other places too.

caroline1233456 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+public class Deadline extends Task {
+    String by;
+
+    public Deadline(String taskName, String date) throws SigmaException {
```
I like how you named the string as taskName. I think it makes the purpose of the variable clear and easy to understand.

caroline1233456 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+            case "mark": {
+                if (tokens.length > 2) { //unknown elements after command
+                    line();
+                    System.out.println(reply_prefix+ "I don't know what you're talking about.");
```
Perhaps the operation "+" should be surrounded by whitespace? I noticed the same issue in several other places too.

caroline1233456 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + "[" + (this.isDone() ? "X" : " ") + "]" + this.getTaskName() + " (By:" + by + ")";
```
I like how you formatted your return statement. I think it makes it easy to read.

caroline1233456 commented on src/main/java/NoDeadlineException.java:
```diff
@@ -0,0 +1,6 @@
+public class NoDeadlineException extends SigmaException {
+    public NoDeadlineException() {
+        super("When's the deadline yo? Add a deadline by using the /by keyword after indicating the task name!");
```
Maybe it would be better if this line is shortened?

caroline1233456 reviewed (commented):
LGTM. Just some nits to fix.

mjyan02 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Sigma {
```
I like how you used nouns for classes/variables and verbs for methods/functions

mjyan02 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+    private String name = "Sigma";
+    private String reply_prefix = name + ": ";
+    
+    private void set_Name(String name) {
```
camelCase should not have underscore. Please follow the basic java coding standard for your other methods and variables as well.

mjyan02 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+        //Token reading solution below inspired by https://www.youtube.com/watch?v=lGHlFaF0F44
+        String[] tokens = reply.split(" ");
+        String command = tokens[0];
+        switch (command) {
```
Please check the correct indentation for switch statements. Case should have the same indentation as the switch line.

mjyan02 commented on src/main/java/Sigma.java:
```diff
@@ -0,0 +1,313 @@
+            case "bye":
+                exit();
+                break;
+            case "list":
```
Perhaps can consider adding line breaks between different cases to improve readability.

mjyan02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+
```
Is this an extra line?

mjyan02 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+public class Deadline extends Task {
```
Javadocs does not seem to have been added yet to any of the files. Do add them in as required by the weekly tasks.

mjyan02 reviewed (commented):
Naming conventions are not always followed for variable and method names. They should be in camelCase without any _ since Java is used instead of Python. 

Boolean variables should be renamed to sound more like booleans i.e. isDone instead of done.

 Javadocs has not been added yet. 

Overall, the code is still readable, but many improvements can be made. All the best!

CXl0l0 committed:
Fix previous Java code standard violations

CXl0l0 committed:
Add feature that save the tasks into hard disk

CXl0l0 committed:
Add date and time recognition for creating deadline tasks

CXl0l0 committed:
Add date and time recognition for creating event tasks

CXl0l0 committed:
Merge branch 'branch-Level-8'

CXl0l0 committed:
Extract classes TaskList and Ui from Sigma.java

CXl0l0 committed:
Merge branch 'branch-A-MoreOOP'

CXl0l0 committed:
Put the classes into their respective packages

CXl0l0 committed:
Merge branch 'branch-A-Packages'

CXl0l0 committed:
Merge branch 'add-gradle-support'

chinpcbenjamin commented on src/sigma/command/Ui.java:
```diff
@@ -0,0 +1,156 @@
+    protected static String replyPrefix = name + ": ";
+
+    private Scanner sc;
+    private TaskList list;
```
Would it be better to change the name from "list" to "taskList"? I think that it could improve code readability further down, as it is more informative to see taskList..... instead of list.....

chinpcbenjamin commented on src/sigma/command/Ui.java:
```diff
@@ -0,0 +1,156 @@
+import sigma.task.TaskList;
+
+public class Ui {
+    protected static String name = "Sigma";
```
I really like this name. 🐐

chinpcbenjamin commented on src/sigma/storage/Storage.java:
```diff
@@ -0,0 +1,100 @@
+        for (int i = 0; i < list.size(); i++) {
+            Task task = list.get(i);
+            String taskName = task.getTaskName();
+            String type = task.getType();
```
Would it be better to call this "taskType" instead of "type"?

chinpcbenjamin commented on src/sigma/task/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+
+public class Deadline extends Task {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private String by;
```
Would it be better to change "by" to "dueDate" or "dueTime"? It informs us what the field is representing. This is especially important considering you are not using the LocalDate class (or its variants) for this field, so it is not clear what the purpose of the field is.

chinpcbenjamin commented on src/sigma/task/Event.java:
```diff
@@ -0,0 +1,60 @@
+
+    public Event(String taskName, boolean done, String from, String to) {
+        super(taskName, done, "E");
+        this.from = from;
+        this.to = to;
```
Would it be better to change "from" to "startDate" or "startTime", and "to" to "endDate" or "endTime"? It informs us what the field is representing. This is especially important considering you are not using the LocalDate class (or its variants) for this field, so it is not clear what the purpose of the field is.

chinpcbenjamin reviewed (commented):
I think the code naming is generally fine, except for some of the issues pointed out. While it may be clear what the purpose is during declaration (as a field), if the variable is only used much further down in the code, it is a little hard to read as I have to trace back and try and find where you declared the variable. Apart from those few code naming issues, I think that most of the other variables are used pretty well (can understand their purpose).

CXl0l0 committed:
Set up gradle and allow building Sigma using gradle

CXl0l0 committed:
Merge branch 'branch-A-Gradle'

CXl0l0 committed:
Add 2 JUnit test classes for ToDo class and Deadline class

CXl0l0 committed:
Merge branch 'branch-A-JUnit'

CXl0l0 committed:
Add JavaDoc documentation to most of the classes and functions.

CXl0l0 committed:
Improve coding standard of the code

CXl0l0 committed:
Add "find" feature for user to find tasks using keywords

CXl0l0 committed:
Merge branch 'branch-A-JavaDoc'

CXl0l0 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/sigma/command/Ui.java
#	src/main/java/sigma/storage/Storage.java
#	src/main/java/sigma/task/Deadline.java
#	src/main/java/sigma/task/Event.java
#	src/main/java/sigma/task/Task.java

CXl0l0 committed:
Merge branch 'branch-Level-9'

CXl0l0 committed:
Make Ui class handles more about the user interface.

CXl0l0 committed:
Add parser class for better OOP

CXl0l0 committed:
Fix and check styles of the main code files and test code files

CXl0l0 committed:
Merge branch 'branch-A-CheckStyle'

CXl0l0 committed:
Add basic GUI to Sigma with no logic implemented

There are no basic GUI for Sigma. User have to rely on terminal
to use Sigma.

Sigma's full potential is limited by a simple UI, better GUI
helps improve user's experience of Sigma.

JavaFX is used to construct a simple GUI for Sigma.

CXl0l0 committed:
Integrate Sigma with GUI

There are no logic in GUI and the response of Sigma only echoes the
user's input.

User couldn't use Sigma's intended functions through the new GUI.

Add new function that takes in input from user through GUI and
output appropriate response.

Inner functions of previous non-GUI iteration wouldn't have to
change much but just return response generated instead.

CXl0l0 committed:
Anchor TextField, Send Button and ScrollPane

These three components in GUI doesn't resize automatically.

No matter how the user resize the window, the components remain the
same size. This would greatly affect the overall user experience of
the GUI.

Set the components' anchor to the AnchorPane. Give the components
constraint to the AnchorPane's side.

Can be done easily through SceneBuilder and achieves the effect
accurately.

CXl0l0 committed:
Tweak Sigma's GUI

The GUI looks boring and doesn't look appealing to the user.

A better GUI would positively affect the user's overall experience
with the app.

Add more colors to the main window.
Style the dialog box.

User won't get as tired looking at a GUI with monotonous color.

CXl0l0 committed:
Add background image to GUI

No background image in main window.

A background image helps with the overall design of the main window.

CXl0l0 committed:
Merge branch 'branch-Level-10'

CXl0l0 committed:
Add assertions into the code

There are no uses of assertion anywhere in the code.

Programmer's mistake is easy to be overlooked if some assumptions
isn't handled well in the code. Catching bugs is easier with
assertions.

Apply use of assertions in parts of the code where certain important
assumptions are made.

CXl0l0 committed:
Refactor codes

Some function's code does not have good SLAP.

A code with better SLAP can help in the future when we are finding
hidden bugs.

Refactor the code through extract method.

CXl0l0 committed:
Merge pull request #2 from CXl0l0/branch-A-Assertions

Add assertions into the code

CXl0l0 committed:
Merge branch 'master' of https://github.com/CXl0l0/ip

CXl0l0 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data/Sigma.txt

CXl0l0 committed:
Merge pull request #3 from CXl0l0/branch-A-CodeQuality

Refactor codes

CXl0l0 committed:
Merge branch 'master' of https://github.com/CXl0l0/ip

CXl0l0 committed:
Give personality to Duke (Sigma)

CXl0l0 committed:
Add feature to edit ToDo tasks

Couldn't edit the name of ToDo tasks.

Ability to edit ToDo task's name would eliminate the tedious
process of creating a new task in order to change the task's
name.

Implement new function whereby Sigma understands the command to edit
the name of ToDo tasks.

CXl0l0 committed:
Add feature to edit Deadline tasks

User couldn't edit Deadline tasks and change the task's name or the
deadline date.

A function to edit deadline tasks eliminates the need to create new
deadline tasks whenever users have to make amendments to Deadline
tasks.

Implement a new feature such that Sigma understands what and how a
user wants to edit a deadline task.

CXl0l0 committed:
Add feature to edit Event tasks

User couldn't edit task name and time period of Event tasks.

Having a feature to edit Event tasks eliminates the tedious steps
user has to go through if they want to make amendments on Event
tasks.

Implement a function such that Sigma understands commands to edit
Event tasks.

CXl0l0 committed:
Merge branch 'branch-A-BCD-Extension'

CXl0l0 committed:
Use Github Copilot to help add JavaDoc on functions

CXl0l0 committed:
Add title and icon image to GUI

CXl0l0 committed:
Add User Guide

There wasn't any user guide for the application.

A user guide covers essential information the user needs to know
in order to use the application effectively.

CXl0l0 committed:
Tweak the user guide

CXl0l0 committed:
Add link for the download of Sigma.jar in user guide

CXl0l0 committed:
Update README.md

CXl0l0 committed:
Fix minor bug in Storage.java

CXl0l0 committed:
Merge branch 'master' of https://github.com/CXl0l0/ip

CXl0l0 committed:
Fix minor bug in Storage
